app/controllers/people_controller.rb

```diff
@@ -117,9 +115,28 @@ def create
 
   def update
     @person = Person.find(params[:id])
-    
+    channel_params = params[:person][:channels_attributes].values
+
+    unless channel_params.empty?
+        channel_params.each do |hash|
```

Comment:
I would be very careful using a reserved word as a var. (hash vs Hash)
Why not params ?